DSCResources/MSFT_xWebsite/MSFT_xWebsite.psm1

```diff
@@ -589,104 +657,38 @@ function Test-WebsiteBindings
         }
     }
 
-    # Assume bindings do not need updating
-    $BindingNeedsUpdating = $false
-
-    <#
-        Currently there is a problem in the LCM where Get-WebBinding short circuts the verbose stream
-        Write-Log can be changed to add the -File switch to log to a directory for troubleshooting.
-        However it's pretty noisy so it's being left in but commented out.
-    #>
-
-    $ActualBindings = Get-WebBinding -Name $Name
-
-    # Format Binding information: Split BindingInfo into individual Properties (IPAddress:Port:HostName)
-    $ActualBindingObjects = @()
-
-    foreach ($ActualBinding in $ActualBindings)
-    {
-        $ActualBindingObjects += Get-WebBindingObject -BindingInfo $ActualBinding
-    }
+    # Assume bindings do not need to be updated
+    $IsUpdateRequired = $false
 
-    #Compare Actual Binding info ($FormatActualBindingInfo) to Desired($BindingInfo)
     try
     {
-        if($BindingInfo.Count -le $ActualBindingObjects.Count)
-        {
-            foreach($binding in $BindingInfo)
-            {
-                $ActualBinding = $ActualBindingObjects | Where-Object -FilterScript {
-                    $_.Port -eq $binding.CimInstanceProperties['Port'].Value
-                }
-                if ($ActualBinding -ne $null)
-                {
-                    if([string]$ActualBinding.Protocol -ne [string]$binding.CimInstanceProperties['Protocol'].Value)
-                    {
-                        Write-Log "Protocol is Incorrect" -File
-                        $BindingNeedsUpdating = $true
-                        break
-                    }
-
-                    if([string]$ActualBinding.IPAddress -ne [string]$binding.CimInstanceProperties['IPAddress'].Value)
-                    {
-                        # Special case where blank IPAddress is saved as "*" in the binding information.
-                        if([string]$ActualBinding.IPAddress -eq '*' -AND [string]$binding.CimInstanceProperties['IPAddress'].Value -eq '')
-                        {
-                            #Do nothing
-                        }
-                        else
-                        {
-                            Write-Log "IP Address Incorrect" -File
-                            $BindingNeedsUpdating = $true
-                            break
-                        }
-                    }
+        $Website = Get-Website | Where-Object -FilterScript {
+            $_.Name -eq $Name
+        }
 
-                    if([string]$ActualBinding.HostName -ne [string]$binding.CimInstanceProperties['HostName'].Value)
-                    {
-                        Write-Log "HostName is incorrect" -File
-                        $BindingNeedsUpdating = $true
-                        break
-                    }
+        # Get-WebBinding may fail in case there are no bindings assigned to a website
+        $CurrentCimBindings = @(
```

Comment:
Same note as above, Piping isn't geting us anything and would be easier to read without it.